dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/UserAccountController.java

```diff
@@ -165,4 +171,15 @@ public void resetPassword(@RequestBody ResetPasswordRequest resetRequest)
 
     log.info("Account restored for user: {}", user.getUsername());
   }
+
+  @PostMapping("/register")
+  public WebMessageResponse register(
+      @RequestBody SelfRegistrationForm selfRegForm, HttpServletRequest request)
+      throws BadRequestException, IOException {
+    log.info("Self registration received");
+    userAccountService.validateUserFormInfo(selfRegForm, request);
```

Comment:
Could we make validation part of the action? What would be the reason to make is into a separate call?

Reply:
hmm, I was trying to break down the copied code into more readable chunks and this seemed to make sense. The validate method doesn't require a transaction so it can be kept separate.
Also, they were left separate so the `AccountController` could use either method if we want to reduce code duplication in that class.